File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/emr/src/main/java/aws/example/emr/DescribeCluster.java
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/
package aws.example.emr;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.emr.EmrClient;
import software.amazon.awssdk.services.emr.model.*;


/*
 *   Ensure that you have setup your development environment, including your credentials.
 *   For information, see this documentation topic:
 *
 *   https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 *
 */
public class DescribeCluster {

    public static void main(String[] args){

        final String USAGE = "\n" +
                "Usage: " +
                "   <clusterId> \n\n" +
                "Where:\n" +
                "   clusterId - The identifier of the cluster to describe. \n\n" ;

        if (args.length != 1) {
              System.out.println(USAGE);
              System.exit(1);
         }


        String clusterId = args[0] ;
        Region region = Region.US_WEST_2;
        EmrClient emrClient = EmrClient.builder()
                .region(region)
                .build();
        describeMyCluster(emrClient, clusterId);

        emrClient.close();
    }

    public static void describeMyCluster(EmrClient emrClient, String clusterId){

        try {
            DescribeClusterRequest clusterRequest = DescribeClusterRequest.builder()
                    .clusterId(clusterId)
                    .build();

            DescribeClusterResponse response = emrClient.describeCluster(clusterRequest);
            System.out.println("The name of the cluster is "+response.cluster().name());

        } catch(EmrException e){
        System.err.println(e.getMessage());
        System.exit(1);
    }
 }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/emr/src/main/kotlin/com/kotlin/emr/DescribeCluster.kt
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.emr

import aws.sdk.kotlin.services.emr.EmrClient
import aws.sdk.kotlin.services.emr.model.DescribeClusterRequest
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args:Array<String>) {

    val usage = """
        Usage:
            <clusterIdVal> 

        Where:
            clusterIdVal - the id of the cluster to describe. 
        """

      if (args.size != 1) {
          println(usage)
          exitProcess(0)
      }

    val clusterIdVal = args[0]
    describeMyCluster(clusterIdVal )
  }

suspend fun describeMyCluster(clusterIdVal: String?) {

     val request = DescribeClusterRequest {
         clusterId = clusterIdVal
     }

     EmrClient { region = "us-west-2" }.use { emrClient ->
        val response = emrClient.describeCluster(request)
        println("The name of the cluster is ${response.cluster?.name}")

     }
}
